File: src/main/kotlin/space/davids_digital/kiri/rest/dto/TelegramChatMetadataDto.kt
package space.davids_digital.kiri.rest.dto

import java.time.ZonedDateTime

data class TelegramChatMetadataDto(
    val lastReadMessageId: Int?,
    val notificationMode: NotificationMode,
    val mutedUntil: ZonedDateTime?,
    val archived: Boolean,
    val pinned: Boolean,
    val enabled: Boolean,
) {
    enum class NotificationMode {
        ALL, ONLY_MENTIONS, NONE
    }
}

File: admin-ui/lib/api/types/TelegramChatMetadataDto.ts
export type TelegramChatMetadataDto = {
    lastReadMessageId: number | null;
    notificationMode: TelegramChatMetadataDto.NotificationMode;
    mutedUntil: number | null;
    archived: boolean;
    pinned: boolean;
    enabled: boolean;
};

export namespace TelegramChatMetadataDto {
    export type NotificationMode = "ALL" | "ONLY_MENTIONS" | "NONE";
}